src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/DefaultModelBindingContext.cs

```diff
@@ -184,6 +188,26 @@ public override ModelBindingResult Result
             }
         }
 
+        private int MaxModelBindingRecursionDepth
+        {
+            get
+            {
+                if (_maxModelBindingRecursionDepth == 0)
+                {
+                    var mvcOptions = ActionContext?.HttpContext?.RequestServices?.GetService<IOptions<MvcOptions>>();
```

Comment:
This is pretty ugly. Would it be reasonable for the calling code to set this value instead of service locatoring?